habitat-lab/habitat/sims/habitat_simulator/sim_utilities.py

```diff
@@ -605,3 +751,76 @@ def within(
         containment_ids.remove(objectA.object_id)
 
     return containment_ids
+
+
+def object_in_region(
+    sim: habitat_sim.Simulator,
+    objectA: Union[
+        habitat_sim.physics.ManagedRigidObject,
+        habitat_sim.physics.ManagedArticulatedObject,
+    ],
+    region: habitat_sim.scene.SemanticRegion,
+    containment_threshold=0.25,
```

Comment:
Inconsistent use of keypoint thresholding across predicates; `within` uses  `keypoint_vote_threshold` to the same effect.

Reply:
Thanks, great point.
I agree that "within" and "object_in_region" are using similar but distinct mechanisms for mostly the same purpose. They could be unified.
For now, merging as-is and we'll decide if a follow-up refactor is desired.